homeassistant/helpers/update_coordinator.py

```diff
@@ -138,16 +142,24 @@ async def async_refresh(self) -> None:
             start = monotonic()
             self.data = await self._async_update_data()
 
-        except asyncio.TimeoutError:
+        except (asyncio.TimeoutError, requests.exceptions.Timeout):
             if self.last_update_success:
                 self.logger.error("Timeout fetching %s data", self.name)
                 self.last_update_success = False
 
-        except aiohttp.ClientError as err:
+        except (aiohttp.ClientError, requests.exceptions.RequestException) as err:
             if self.last_update_success:
                 self.logger.error("Error requesting %s data: %s", self.name, err)
                 self.last_update_success = False
 
+        except urllib.error.URLError as err:
+            if self.last_update_success:
+                if err.reason == "timed out":
```

Comment:
I wish there was a cleaner way to trap this. I couldn't find one.

Reply:
Yeah, ditto.